File: sdk-ios/Sources/StackhouseBrain.swift
import Foundation

public struct ChatRequest: Codable {
    public let sessionId: String
    public let message: String
    
    enum CodingKeys: String, CodingKey {
        case sessionId = "session_id"
        case message
    }
}

public struct AgentTask: Codable {
    public let goal: String
    public let agentType: String?
    
    enum CodingKeys: String, CodingKey {
        case goal
        case agentType = "agent_type"
    }
}

public class StackhouseBrain {
    private let url: URL
    private let session = URLSession.shared
    
    public init(url: URL) {
        self.url = url.appendingPathComponent("v1/brain")
    }
    
    public func chat(request: ChatRequest) async throws -> [String: Any]? {
        var req = URLRequest(url: url.appendingPathComponent("chat"))
        req.httpMethod = "POST"
        req.addValue("application/json", forHTTPHeaderField: "Content-Type")
        req.httpBody = try JSONEncoder().encode(request)
        
        let (data, _) = try await session.data(for: req)
        let json = try JSONSerialization.jsonObject(with: data, options: []) as? [String: Any]
        return json?["data"] as? [String: Any]
    }
}

File: sdk-flutter/lib/stackhouse_brain.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

class ChatRequest {
  final String sessionId;
  final String message;

  ChatRequest({required this.sessionId, required this.message});

  Map<String, dynamic> toJson() => {
    'session_id': sessionId,
    'message': message,
  };
}

class AgentTask {
  final String goal;
  final String? agentType;

  AgentTask({required this.goal, this.agentType});

  Map<String, dynamic> toJson() => {
    'goal': goal,
    'agent_type': agentType,
  };
}

class StackhouseBrain {
  final String url;
  
  StackhouseBrain(this.url);

  Future<Map<String, dynamic>> chat(ChatRequest request) async {
    final response = await http.post(
      Uri.parse('$url/v1/brain/chat'),
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode(request.toJson()),
    );
    
    if (response.statusCode == 200) {
      return jsonDecode(response.body);
    } else {
      throw Exception('Failed to chat with AI Brain');
    }
  }

  Future<Map<String, dynamic>> submitTask(AgentTask task) async {
    final response = await http.post(
      Uri.parse('$url/v1/agent/task'),
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode(task.toJson()),
    );
    
    if (response.statusCode == 200) {
      return jsonDecode(response.body);
    } else {
      throw Exception('Failed to submit agent task');
    }
  }
}
